src/main/java/frc/robot/subsystems/swerve/SwerveCommandsBuilder.java

```diff
@@ -144,6 +146,22 @@ public Command driveByState(DoubleSupplier xSupplier, DoubleSupplier ySupplier,
 		).withName("Drive with state");
 	}
 
+	public Command followPathOrDriveToPathEnd(Supplier<Pose2d> currentPose, PathPlannerPath path) {
+		return new ConditionalCommand(
```

Comment:
this needs some documentation or simplification as it is very hard to understand